webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/SecretApi.scala

```diff
@@ -0,0 +1,165 @@
+
+  def getLiftEndpoints(): List[LiftApiModule] = {
+    API.endpoints.map(
+      e => e match {
+        case API.GetAllSecret => GetAllSecret
+        case API.GetSecret    => GetSecret
+        case API.AddSecret    => AddSecret
+        case API.UpdateSecret => UpdateSecret
+        case API.DeleteSecret => DeleteSecret
+      }
+    )
+  }
+
+  object GetAllSecret extends LiftApiModule0 {
+    val schema        = API.GetAllSecret
+    val restExtractor = restExtractorService
+
+    def process0(version: ApiVersion, path: ApiPath, req: Req, params: DefaultParams, authzToken: AuthzToken): LiftResponse = {
+      implicit val prettify = params.prettify
+      implicit val action   = schema.name
+
+      val res = for {
+        secrets <- secretService.getAllSecret.toBox
+      } yield {
+        JArray(secrets.map(Secret.serializeSecret))
+      }
+
+      RestUtils.response(restExtractor, "secrets", None)(res, req, "Error when trying to get all secrets")
+    }
+  }
+
+  object GetSecret extends LiftApiModule {
+    val schema        = API.GetSecret
+    val restExtractor = restExtractorService
+
+    def process(version: ApiVersion, path: ApiPath, id: String, req: Req, params: DefaultParams, authzToken: AuthzToken): LiftResponse = {
+      implicit val prettify = params.prettify
+      implicit val action   = schema.name
+
+      val res = for {
+        secret <- secretService.getSecretById(id).notOptional(s"Could not find secret with `${id}` name").toBox
+      } yield {
+        Secret.serializeSecret(secret)
+      }
+
+      RestUtils.response(restExtractor, "secrets", None)(res, req, s"Error when trying to get secret `${id}` value")
+    }
+  }
+
+  object AddSecret extends LiftApiModule0 {
+    val schema        = API.AddSecret
+    val restExtractor = restExtractorService
+
+    def process0(version: ApiVersion, path: ApiPath, req: Req, params: DefaultParams, authzToken: AuthzToken): LiftResponse = {
+      implicit val prettify = params.prettify
+      implicit val action = schema.name
+
+      val res = for {
+        secret <- restExtractorService.extractSecret(req)
+        _      <- secretService.addSecret(secret, "Add a secret").toBox
+      } yield {
+        Secret.serializeSecret(secret)
+      }
+
+      RestUtils.response(restExtractor, "secret", None)(res, req, "Error when trying to add a secret")
+    }
+  }
+
+  object DeleteSecret extends LiftApiModule {
+    val schema        = API.DeleteSecret
+    val restExtractor = restExtractorService
+
+    def process(version: ApiVersion, path: ApiPath, id: String, req: Req, params: DefaultParams, authzToken: AuthzToken): LiftResponse = {
+      implicit val prettify = params.prettify
+      implicit val action = schema.name
+
+      val res = for {
+        _ <- secretService.deleteSecret(id, "Delete a secret").toBox
+      } yield {
+        JString(id)
+      }
+
+      RestUtils.response(restExtractor, "secretName", None)(res, req, "Error when trying to delete a secret")
```

Comment:
you should always send back the whole secret structure with the same container than the rest of the api